src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfT.cs

```diff
@@ -102,18 +100,14 @@ public override bool GetMemberAndWriteJson(object obj, ref WriteStack state, Utf
             bool success;
             T value = Get!(obj);
 
-            if (value == null)
+            if (IgnoreDefaultValuesOnWrite && (
+                default(T) is null ? value is null : EqualityComparer<T>.Default.Equals(default, value)))
             {
-                Debug.Assert(s_defaultValue == null && Converter.CanBeNull);
-
-                success = true;
-                if (!IgnoreDefaultValuesOnWrite)
+                if (default(T) is null)
                 {
-                    if (!Converter.HandleNull)
-                    {
-                        writer.WriteNull(EscapedName.Value);
-                    }
-                    else
+                    Debug.Assert(Converter.CanBeNull);
+
+                    if (Converter.HandleNull)
```

Comment:
Why is it named as `HandleNull`, but not `HandlesNull`? Is it a short form of `ShouldHandleNull`?